src/python/grapl_analyzerlib/grapl_analyzerlib/nodes/network_connection.py

```diff
@@ -0,0 +1,160 @@
+            EdgeT(NetworkConnectionSchema, IpPortSchema, EdgeRelationship.ManyToOne),
+            "network_connections_from",
+        )
+    }
+
+
+class NetworkConnectionSchema(EntitySchema):
+    def __init__(self):
+        super(NetworkConnectionSchema, self).__init__(
+            default_network_connection_properties(),
+            default_network_connection_edges(),
+            lambda: NetworkConnectionView,
+        )
+
+    @staticmethod
+    def self_type() -> str:
+        return "NetworkConnection"
+
+
+class NetworkConnectionQuery(EntityQuery[IPPV, IPPQ]):
+
+    @with_int_prop('port')
+    def with_port(
+            self,
+            *,
+            eq: Optional["IntOrNot"] = None,
+            gt: Optional["IntOrNot"] = None,
+            ge: Optional["IntOrNot"] = None,
+            lt: Optional["IntOrNot"] = None,
+            le: Optional["IntOrNot"] = None,
+    ):
+        pass
+
+    @with_str_prop('ip_address')
+    def with_ip_address(
+            self,
+            *,
+            eq: Optional["StrOrNot"] = None,
+            contains: Optional["OneOrMany[StrOrNot]"] = None,
+            starts_with: Optional["StrOrNot"] = None,
+            ends_with: Optional["StrOrNot"] = None,
+            regexp: Optional["OneOrMany[StrOrNot]"] = None,
+            distance_lt: Optional[Tuple[str, int]] = None,
+    ):
+        pass
+
+    def with_inbound_network_connection_to(self, *inbound_network_connection_to):
+        return self.with_to_neighbor(IpPortQuery, 'inbound_network_connection_to',
+                                     'network_connections_from', inbound_network_connection_to)
+
+
+    @staticmethod
+    def extend_self(*types):
+        for t in types:
+            method_list = [
+                method for method in dir(t) if method.startswith("__") is False
+            ]
+            for method in method_list:
+                setattr(NetworkConnectionQuery, method, getattr(t, method))
+        return type("NetworkConnectionQuery", types, {})
+
+    @classmethod
+    def node_schema(cls) -> "Schema":
+        return NetworkConnectionSchema()
+
+
+class NetworkConnectionView(EntityView[IPPV, IPPQ]):
+    queryable = NetworkConnectionQuery
+
+    def __init__(
+            self,
+            uid: str,
+            node_key: str,
+            graph_client: Any,
+            node_types: Set[str],
+            port: Optional[int] = None,
+            ip_address: Optional[str] = None,
+            **kwargs,
+    ):
+        super().__init__(uid, node_key, graph_client, **kwargs)
+        self.node_types = set(node_types)
+
+        self.port = port
+        self.ip_address = ip_address
+
+    def get_port(self, cached=True):
+        return self.get_int('port', cached=cached)
+
+    def get_ip_address(self, cached=True):
+        return self.get_str('ip_address', cached=cached)
+
+    def get_inbound_network_connection_to(self, *inbound_network_connection_to, cached=False):
```

Comment:
for edge names, thoughts on making them a str enum somewhere? (I had to double-check these were the same and no typos with line 81 for example)